cli/src/config.rs

```diff
@@ -434,6 +435,41 @@ fn read_config_path(config_path: &Path) -> Result<config::Config, config::Config
         .build()
 }
 
+#[cfg(target_os = "macos")]
+fn config_dir() -> Option<PathBuf> {
+    // Define a closure to check if config file exists in a directory
+    let check_config = |dir: &Path| {
+        let mut config_file = dir.to_owned();
+        config_file.push("jj/config.toml");
+        if config_file.exists() {
+            Some(dir.to_owned())
+        } else {
+            None
+        }
+    };
+
+    // Try getting config directory from Xdg base strategy first.
+    let path = etcetera::base_strategy::Xdg::new()
+        .ok()
+        .and_then(|strategy| check_config(&strategy.config_dir()));
+
+    // If config directory is not found from Xdg, try Apple instead. `Application
+    // Support` is really the data_dir, so get that.
+    path.or_else(|| {
+        etcetera::base_strategy::Apple::new()
+            .ok()
+            .and_then(|strategy| check_config(&strategy.data_dir()))
+    })
```

Comment:
Hmm, actually this will result in the config dir still being `~/Library/Application Support/` if you don't have a config yet.

Reply:
Hm I can't reproduce that. On my machine, if I don't have a config either place, it creates `$HOME/.jjconfig.toml`, which is probably not what we want either.